### ISSUE TITLE
`role-supports-aria-props`, `no-noninteractive-element-interactions`: Cannot read property 'expression' of undefined

### ISSUE THREAD

**Issue by AriPerkkio** [jsx-eslint/eslint-plugin-jsx-a11y#770]
Hello, `role-supports-aria-props` rule seems to crash in certain cases. This issue was spotted by automated CI run - it is not blocking my development or anything. https://github.com/AriPerkkio/eslint-remote-tester/runs/1828268606?check_suite_focus=true

This was the first time I was able to crash this plugin 🎉 . For the past three months it has always successfully linted +10K repositories. This is one of the most stable plugin in [the test set](https://github.com/AriPerkkio/eslint-remote-tester/tree/master/ci/plugin-configs). 
 
> `[INFO/STATUS] Repositories (12561/14496)`

```
"eslint-plugin-jsx-a11y": "^6.4.1"
```

```json
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2020,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "settings": {
    "react": {
      "version": "16.13.1"
    }
  },
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/role-supports-aria-props": "error"
  }
}
```

### Minimal repro:

- Conditional chaining
- Type assertion

```tsx
import React from "react";

export default function Repro({ images }) {
  return <img src={images?.footer as string} />;
}
```

```
TypeError: Cannot read property 'expression' of undefined
Occurred while linting <text>:46
    at extract (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/values/expressions/index.js:171:20)
```

<details>
  <summary>Crash reports from real projects</summary>

## Rule: role-supports-aria-props

- Message: `Cannot read property 'expression' of undefined Occurred while linting <text>:46`
- Path: `expo/snack/website/src/client/components/AssetViewer.tsx`
- [Link](https://github.com/expo/snack/blob/HEAD/website/src/client/components/AssetViewer.tsx#L46)

```tsx
  44 |         <div className={css(styles.inner)}>
  45 |           {isImage(selectedFile.toLowerCase()) ? (
> 46 |             <img
  47 |               onLoad={this._handleImageLoad}
  48 |               className={css(styles.preview)}
  49 |               src={files[selectedFile]?.contents as string}
```

```
TypeError: Cannot read property 'expression' of undefined
Occurred while linting <text>:46
    at extract (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/values/expressions/index.js:171:20)
    at Object.extractValueFromChainExpression [as ChainExpression] (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/values/expressions/ChainExpression.js:18:10)
    at Object.extract [as JSXExpressionContainer] (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/values/expressions/index.js:201:21)
    at getValue (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/values/index.js:51:27)
    at extractValue (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/getPropValue.js:24:12)
    at getPropValue (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/jsx-ast-utils/lib/getPropValue.js:41:10)
    at /home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/eslint-plugin-jsx-a11y/lib/rules/role-supports-aria-props.js:64:45
    at Array.forEach (<anonymous>)
    at JSXOpeningElement (/home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/eslint-plugin-jsx-a11y/lib/rules/role-supports-aria-props.js:62:25)
    at /home/runner/work/eslint-remote-tester/eslint-remote-tester/ci/node_modules/eslint/lib/linter/safe-emitter.js:45:58
```

</details>

**Comment by ljharb** [comment]
This seems to work fine with the default parser, and we don't yet have any tests set up to run on any non-default parsers.

**Comment by ljharb** [comment]
What version of jsx-ast-utils are you using?

**Comment by zomars** [comment]
> What version of jsx-ast-utils are you using?

We're not using `jsx-ast-utils` directly we're using it by [eslint-config-next](https://github.com/vercel/next.js/tree/canary/packages/eslint-config-next). And they're using `"eslint-plugin-jsx-a11y": "^6.5.1"` AFAIK.

**Comment by ljharb** [comment]
@zomars right but what does `npm ls jsx-ast-utils` print out?

**Comment by zomars** [comment]
<img width="354" alt="image" src="https://user-images.githubusercontent.com/3504472/154591557-734c6196-7166-484f-9aad-000f4b1ca422.png">

**Comment by ljharb** [comment]
The next version of jsx-ast-utils should theoretically fix this.

**Comment by ljharb** [comment]
jsx-ast-utils v3.3.1 is released, and should resolve this.
